plugins/pur/core/src/test/java/org/pentaho/di/repository/pur/PurRepositoryStressTest.java

```diff
@@ -0,0 +1,587 @@
+    Calendar calendar = mock( Calendar.class );
+
+    purRepository.save( rei, "vComment", calendar, mock( ProgressMonitorListener.class ), false );
+  }
+
+  public void testLockSaveSlaverSchema() throws Exception {
+    ObjectId objId = mock( ObjectId.class );
+    doReturn( "id1" ).when( objId ).getId();
+
+    SlaveServer rei = mock( SlaveServer.class );
+    doReturn( RepositoryObjectType.SLAVE_SERVER ).when( rei ).getRepositoryElementType();
+    doReturn( "slaveID" ).when( rei ).getName();
+    doReturn( objId ).when( rei ).getObjectId();
+    doReturn( rei ).when( rei ).clone();
+    Calendar calendar = mock( Calendar.class );
+
+    purRepository.save( rei, "vComment", calendar, mock( ProgressMonitorListener.class ), false );
+  }
+
+  public void testLockGetDatabaseMetaParentFolderId() {
+    assertNotNull( purRepository.getDatabaseMetaParentFolderId() );
+  }
+
+  public void testLockUndeleteObject() throws Exception {
+    ObjectId objId = mock( ObjectId.class );
+    doReturn( "id1" ).when( objId ).getId();
+    RepositoryElementMetaInterface rem = mock( RepositoryElementMetaInterface.class );
+    doReturn( objId ).when( rem ).getObjectId();
+    purRepository.undeleteObject( rem );
+
+    verify( rem, times( 1 ) ).getObjectId();
+    verify( objId, times( 1 ) ).getId();
+  }
+
+  public void testLockGetObjectInformation() throws Exception {
+    ObjectId objId = mock( ObjectId.class );
+    doReturn( "idnull" ).when( objId ).getId();
+
+    assertNull( purRepository.getObjectInformation( objId, null ) );
+  }
+
+  public void testLockLoadJob2() throws Exception {
+    ObjectId objId = mock( ObjectId.class );
+    doReturn( "id1" ).when( objId ).getId();
+
+    assertNotNull( purRepository.loadJob( objId, "v3" ) );
+    verify( objId, times( 2 ) ).getId();
+  }
+
+  public void testLockLoadTransformation2() throws Exception {
+    ObjectId objId = mock( ObjectId.class );
+    doReturn( "id1" ).when( objId ).getId();
+
+    assertNotNull( purRepository.loadTransformation( objId, "v3" ) );
+    verify( objId, times( 2 ) ).getId();
+  }
+
+  public void testLockLoadGetJobsUsingDatabase() throws Exception {
+    ObjectId objId = mock( ObjectId.class );
+    doReturn( "id1" ).when( objId ).getId();
+
+    assertNotNull( purRepository.getJobsUsingDatabase( objId ) );
+    verify( objId, times( 1 ) ).getId();
+  }
+
+  public void testLockLoadGetTransformationsUsingDatabase() throws Exception {
+    ObjectId objId = mock( ObjectId.class );
+    doReturn( "id1" ).when( objId ).getId();
+
+    assertNotNull( purRepository.getTransformationsUsingDatabase( objId ) );
+    verify( objId, times( 1 ) ).getId();
+  }
+
+  public void testLockGetChildren() {
+    purRepository.getChildren( "/folder1/folder2/", "" );
+  }
+
+  private class Flow implements Runnable {
+    boolean success = false;
+    Throwable t = null;
+    int nr;
+
+    public Flow( int nr ) {
+      this.nr = nr;
+    }
+
+    @Override
+    public void run() {
```

Comment:
I like the idea behind this test list. You should however make the tests run in a random order on each thread.

Reply:
I thought to do it also, but at the time I was doing the dev I don't know if was to much. Ty for the feedback, I'll add random order for testing sample list.

Reply:
@lucboudreau  just updated PR with tests being selected by a random order